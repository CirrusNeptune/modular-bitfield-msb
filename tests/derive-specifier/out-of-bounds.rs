use modular_bitfield_msb::prelude::*;

#[bitfield(filled = false)]
#[derive(BitfieldSpecifier, Debug)]
pub struct Header {
    a: B1,
    b: B128,
}

fn main() {}
