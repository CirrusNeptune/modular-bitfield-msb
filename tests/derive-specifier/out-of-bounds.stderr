error[E0277]: the trait bound `False: SpecifierHasAtMost128Bits` is not satisfied
   --> $DIR/out-of-bounds.rs:4:1
    |
4   | / #[derive(BitfieldSpecifier, Debug)]
5   | | pub struct Header {
6   | |     a: B1,
7   | |     b: B128,
8   | | }
    | |_^ the trait `SpecifierHasAtMost128Bits` is not implemented for `False`
    |
   ::: $WORKSPACE/src/private/checks.rs
    |
    |       <Self::CheckType as DispatchTrueFalse>::Out: SpecifierHasAtMost128Bits,
    |                                                    ------------------------- required by this bound in `CheckSpecifierHasAtMost128Bits`
